tests/integration/kubernetes/gha-run.sh

```diff
@@ -554,11 +554,24 @@ function cleanup_nydus_snapshotter() {
 	else
 		kubectl_retry delete --ignore-not-found -f "misc/snapshotter/base/nydus-snapshotter.yaml"
 	fi
-	sleep 180s
+
+	# The extensive sleep durations lead to certain tests timing out. 
+	# Commenting the sleep for archival purposes for now.
+	#sleep 180s
+
 	kubectl_retry delete --ignore-not-found -f "misc/snapshotter/nydus-snapshotter-rbac.yaml"
-	kubectl_retry get namespace nydus-system -o json | jq 'del(.spec.finalizers)' | kubectl_retry replace --raw "/api/v1/namespaces/nydus-system/finalize" -f - || true
+	kubectl get namespace nydus-system -o json | jq 'del(.spec.finalizers)' | kubectl replace --raw "/api/v1/namespaces/nydus-system/finalize" -f - || true
 	popd
-	sleep 30s
+
+	# The extensive sleep durations lead to certain tests timing out. 
+	# Commenting the sleep for archival purposes for now.
+	#sleep 30s
+
+	# Deleting the nydus snapshotter directory
+	if [ -d "${nydus_snapshotter_install_dir}" ]; then
+		rm -rf "${nydus_snapshotter_install_dir}"
+	fi
```

Comment:
Don't do this, as the cleanup is also called during install, to remove any leftover installation.

Reply:
We were discussing this in the slack thread, but the `kubectl_retry` call to patch the finalizer in the cleanup was causing an extended delay because it wasn't finding anything in the `nydus-system` namespace (already deleted) and was retrying the check 5 times 15 secs each time. This command is removed now in your revised commit, and I'm guessing the other `kubectl_retry` delete calls that have the `--ignore-not-found` flag won't have this issue, so maybe no hanging in the cleanup anymore. However, everything in the cleanup is skipped based on the initial check for the directory, and it was cleaned up. So my thought was that it would be nice to keep this here to streamline the deploy in case the cleanup call takes some time.

Wouldn't it be best to just completely clean it up and remove the directory in the cleanup method?

Reply:
So, let's think about what's happening here.
1. We call the method to deploy nydus snapshotter
2. The method itself will download nydus
3. And call the clean up function
4. Which will then delete the directory if this is added where it was added
5. Snapshot deployment will fail as the directory was removed

This is for sure **not** what we want.

Reply:
My bad, I see your point. This is the wrong place. I still think we should remove it as part of a final cleanup step (part not to be run during deployment). Will look into the right place to put it.